### PR TITLE
Rotate

### PR DESCRIPTION
**Pull request by perlindgren**
A small rewrite allowing flip and rotate of tiles according to x, y, and d (anti diagonal). It works with the import from "tiled" (map editor). 

Limitations: D flipping is currently supported only for "square" maps.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "bevy_ecs_tilemap"
 description = "A tilemap rendering plugin for bevy which is more ECS friendly by having an entity per tile."
-version = "0.4.0"
+version = "0.4.1"
```
Comment (StarArawn) [inline Cargo.toml:4]: Let's roll back this change. I push releases manually there is no automated process for that and what's in master is a 0.5 not a 0.4.1 release.

File: src/render/square-tilemap.vert [hunk 2]
```diff
@@ -27,11 +27,11 @@ void main() {
     vec2 position = Vertex_Position.xy;
     
     vec2 positions[4] = vec2[4](
-        vec2(position.x, position.y),
-        vec2(position.x, position.y + 1.0),
-        vec2(position.x + 1.0, position.y + 1.0),
-        vec2(position.x + 1.0, position.y)
-    );
+            vec2(position.x, position.y),
```
Comment (StarArawn) [inline src/render/square-tilemap.vert:30]: Formatting looks a little off here.

**Review by StarArawn — CHANGES_REQUESTED** [review]

File: src/render/square-tilemap.vert (hunk 2 above)
Comment (perlindgren) [reply src/render/square-tilemap.vert]: Yea, I updated my rotate branch, let me know what you think. (I also added another small comment for completeness)

File: Cargo.toml (hunk 1 above)
Comment (perlindgren) [reply Cargo.toml]: I fixed it in my rotate branch.

**Review by StarArawn — APPROVED** [review]

**Pull request merged**